pagination/pager.go

```diff
@@ -83,9 +84,10 @@ func (p Pager) fetchNextPage(url string) (Page, error) {
 	return p.createPage(remembered), nil
 }
 
-// EachPage iterates over each page returned by a Pager, yielding one at a time to a handler function.
-// Return "false" from the handler to prematurely stop iterating.
-func (p Pager) EachPage(handler func(Page) (bool, error)) error {
+// EachPageWithContext iterates over each page returned by a Pager, yielding
+// one at a time to a handler function. Return "false" from the handler to
+// prematurely stop iterating.
+func (p Pager) EachPageWithContext(ctx context.Context, handler func(context.Context, Page) (bool, error)) error {
```

Comment:
This could be:

```Go
func (p Pager) EachPageWithContext(ctx context.Context, handler func(Page) (bool, error)) error {
```

... if we want the caller to embed an external context with a closure.